src/meta/src/barrier/recovery.rs

```diff
@@ -385,13 +387,17 @@ impl GlobalBarrierManager {
                         tracing::Span::current(), // recovery span
                     ));
 
-                    let mut node_to_collect = control_stream_manager
-                        .inject_barrier(command_ctx.clone(), info.existing_table_ids())?;
+                    let mut node_to_collect = control_stream_manager.inject_barrier(
+                        &command_ctx,
+                        &info.fragment_infos,
+                        Some(&info.fragment_infos),
```

Comment:
Why we pass the same info twice

Reply:
The two parameters have different semantics. It just happens to be the same one in recovery.